lib/urlwatch/reporters.py

```diff
@@ -1166,3 +1166,41 @@ def submit(self):
             'priority': self.config['priority'],
             'title': self.config['title'],
         })
+
+
+class NtfyReporter(TextReporter):
+    """Send messages to a ntfy server"""
+
+    __kind__ = 'ntfy'
+
+    def submit(self):
+        topic_url = self.config['topic_url']
+        headers = {}
+        config_priorities = self.config.get("priorities", {})
+        if priority := config_priorities.get("default"):
+            headers['Priority'] = priority
+        if authorization := self.config.get('authorization'):
+            headers['Authorization'] = authorization
+
+        for job_state in self.report.get_filtered_job_states(self.job_states):
+            title = f'urlwatch {job_state.verb.upper()}: {job_state.job.pretty_name()}'
+            content = self._format_content(job_state)
+            # requests does not handle utf-8-encoded strings in
+            # headers properly, so they are encoded beforehand
+            job_headers = {
+                **headers,
+                'Title': title.encode('utf-8'),
+            }
+            if priority := config_priorities.get(job_state.verb):
+                job_headers["Priority"] = priority
+            if job_state.job.location_is_url():
+                job_headers['Actions'] = f"view, Open URL, {job_state.job.get_location()}, clear=true".encode('utf-8')
+            r = requests.post(
+                topic_url,
+                headers=job_headers,
+                data=content.encode(encoding='utf-8') if content is not None else None,
+            )
+            try:
+                r.raise_for_status()
+            except:
```

Comment:
Possibly limit the catching to `requests.exceptions.HTTPError` and `requests.exceptions.RequestException`? Don't want to [catch them all](https://wiki.c2.com/?PokemonExceptionHandling).